### PR TITLE
[Snyk] Security upgrade onfido-sdk-ui from 13.6.1 to 14.15.0

### PR DESCRIPTION
**Pull request by DylanDylann**
### Details

- Changes to the following files to upgrade the vulnerable dependencies to a fixed version:
         - package.json
         - package-lock.json
         
**Vulnerabilities that will be fixed**
**With an upgrade:**

Severity | Priority Score (*) | Issue | Breaking Change | Exploit Maturity
-- | -- | -- | -- | --
  | 631/1000Why? Proof of Concept exploit, Has a fix available, CVSS 6.2 | Missing Release of Resource after Effective LifetimeSNYK-JS-INFLIGHT-6095116 | Yes | Proof of Concept

(*) Note that the real score may have changed since the PR was raised.

Check the changes in this PR to ensure they won't cause issues with your project.

### Fixed Issues

$ https://github.com/Expensify/App/issues/37076
PROPOSAL: NA

### Tests

Ensure Onfido is not broken for the flows where we add a bank account
- [x] Verify that no errors appear in the JS console

### QA Steps

Ensure Onfido is not broken for the flows where we add a bank account
- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/141406735/4a4f3b45-9589-49a2-834b-40955a7f37cf

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/141406735/26729700-552e-419d-9bc5-63e81ebf3518

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/141406735/46a8a97f-7dd8-4247-be65-1438e629e109

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/141406735/257582cd-a574-4ef1-b058-76dddf448112

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/141406735/7718ff5d-c5cb-4372-bca4-ba736c73b303

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/141406735/6920e42c-6b9d-45bb-9c78-ea47dd01effb

</details>

### CODE REVIEW

**Comment by DylanDylann** [comment]
@marcochavezf Do we need another C+ contributor here?

**Comment by DylanDylann** [comment]
@marcochavezf The PR is ready for review

**Comment by DylanDylann** [comment]
Validate Github Actions also fails in other [PR](https://github.com/Expensify/App/pull/37679) that change package.json file

**Comment by marcochavezf** [comment]
> @marcochavezf Do we need another C+ contributor here?

Yeah, @fedirjh could you review this PR? The more 👀 the better

**Review by fedirjh — CHANGES_REQUESTED** [review]
@DylanDylann Can you please include the changelog in the PR description? 

It seems that `useMemoryHistory` has been deprecated as well.

File: src/components/Onfido/BaseOnfidoWeb.js [hunk 1]
```diff
@@ -91,13 +91,6 @@ function initializeOnfido({sdkToken, onSuccess, onError, onUserExit, preferredLo
             Log.hmmm('Onfido error', {errorType, errorMessage});
             onError(errorMessage);
         },
-        onUserExit: (userExitCode) => {
-            Log.hmmm('Onfido user exits the flow', {userExitCode});
-            onUserExit(userExitCode);
-        },
```
[the remark this answers is not in the document]
Comment (DylanDylann) [reply src/components/Onfido/BaseOnfidoWeb.js:97]: Thanks. I updated but I can't find any way to trigger onUserExit (old) and onError with user_consent_denied type

**Comment by DylanDylann** [comment]
@fedirjh Bump on this one

**Review by fedirjh — COMMENTED** [review]
Looks good and test well. @DylanDylann Could you please merge main ?

**Comment by DylanDylann** [comment]
@fedirjh All done

**Review by fedirjh — APPROVED** [review]

File: src/components/Onfido/BaseOnfidoWeb.js [hunk 2]
```diff
@@ -86,18 +85,15 @@ function initializeOnfido({sdkToken, onSuccess, onError, onUserExit, preferredLo
...
         },
         onError: (error) => {
-            const errorMessage = lodashGet(error, 'message', CONST.ERROR.UNKNOWN_ERROR);
             const errorType = lodashGet(error, 'type');
+            if (errorType === CONST.WALLET.ERROR.ONFIDO_USER_CONSENT_DENIED) {
+                onUserExit();
+                return;
+            }
+            const errorMessage = lodashGet(error, 'message', CONST.ERROR.UNKNOWN_ERROR);
             Log.hmmm('Onfido error', {errorType, errorMessage});
```
Comment (marcochavezf) [inline src/components/Onfido/BaseOnfidoWeb.js:94]: Can we move this line above? So we can log something even if the error is `ONFIDO_USER_CONSENT_DENIED`?

**Review by marcochavezf — CHANGES_REQUESTED** [review]
LGTM, just a minor change

File: src/components/Onfido/BaseOnfidoWeb.js (hunk 2 above)
Comment (DylanDylann) [reply src/components/Onfido/BaseOnfidoWeb.js:94]: Updating

**Comment by DylanDylann** [comment]
@marcochavezf Updated. Passed all tests

**Review by marcochavezf — APPROVED** [review]
Cool, thanks guys!

**Comment by DylanDylann** [comment]
@mkhutornyi Any problem about my change?

**Comment by DylanDylann** [comment]
This PR didn't add these changes

**Comment by DylanDylann** [comment]
I am not sure why I missed that while executing npm install. I created a new [PR](https://github.com/Expensify/App/pull/38110) to remove that. Thanks for your reminder

cc @fedirjh @marcochavezf

**Comment by VickyStash** [comment]
@DylanDylann Have you had a chance to take a look at my [comment](https://github.com/Expensify/App/pull/37713#issuecomment-1991032873)?

**Comment by DylanDylann** [comment]
@VickyStash I don't see any problem with the link color (even though when hovering). Could you help to check again?

https://github.com/Expensify/App/assets/141406735/81034c28-42c2-4a8d-97f5-b687d498f1bc

**Comment by DylanDylann** [comment]
About the Back button: Currently we are using the default back button ([ref](https://jsfiddle.net/gh/get/library/pure/onfido/onfido-sdk-ui/tree/master/demo/fiddle/)) and I don't see any way to change the UI of the back button (this is UI customization [docs](https://github.com/onfido/onfido-sdk-ui/blob/825d3a1facb6a6b60ae6d67d4a76056f38b2de97/UI_CUSTOMIZATION.md)). It seems the default UI is changed in the new version

**Comment by VickyStash** [comment]
@DylanDylann Links look okay for me as well now, but I still see the black border
image

**Comment by VickyStash** [comment]
@DylanDylann oh, nvm, I saw your comment in my PR 👌

**Pull request merged**